### DISCUSSION TITLE
Magick++ : How to center Caption with Best Fit?

### DISCUSSION THREAD

**Discussion by bangusi** [ImageMagick/ImageMagick#2628] [Help] [answered]
I am on Windows:
This command will center the text and make it large enough to fit the image. Exactly the way I want
`magick convert -size 1440x720 -background black -font "c:\windows\fonts\Times.ttf" -fill white -gravity center caption:"Hello Centered" hello.png`

Using the C++ API is what I really want to accomplish the same and this is my code that does not produce the desired result. 
The text it placed at the NW location using a very small font size

```
	Magick::Image hello(Magick::Geometry(1440, 720), Magick::Color("black"));
	std::string font = "c:\\windows\\fonts\\Times.ttf";
	hello.font(font);
	hello.fillColor(Magick::Color("white"));
	hello.read("CAPTION:Hello World!");
	hello.write("hello.png");
```

**Comment by snibgo** [comment, marked as answer]
You need:
```
    hello.textGravity( MagickCore::CenterGravity );
```
Auto-pointsize works fine for me, with IM v7.0.8-64. What version do you have?

**Comment by bangusi** [comment]
Being very new to Imagemagick it appears I am mixing up the libraries ( imagemagick and graphicsmagick ).
I have the development libraries installed for graphicsmagikc but not for imagemagick. The C++ header file appears to be similar but the behaviours of the two libraries are clearly different

**Comment by snibgo** [comment]
Yes, GM and IM are different systems. (GM was a fork from IM, many years ago.) I suggest you use the same system for command line (CLI) and programming (API).

I don't know GM, and can't help you with that.
